### PR TITLE
access_all_profiles permission wasn't working as expected on mri_viol module. [rm8815]

### PR DESCRIPTION
**Pull request by tedstrauss**

### CODE REVIEW

File: modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc [hunk 1]
```diff
@@ -147,6 +147,13 @@ class NDB_Menu_Filter_Form_mri_violations extends NDB_Menu_Filter_Form
         else{
             $useProjects = true;
         }
+        // Get sites
+        $sitesCond = '';
+        if (!$user->hasPermission('access_all_profiles') 
+            && !is_null($user->getData('CenterID'))
+        ) {
+            $sitesCond = " AND p.CenterID = '".$user->getData('CenterID')."' ";
```
Comment (xlecours) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: Remark: p.CenterID is a numeric column and not text. The ' characters should be removed.
But [MySQL](http://dev.mysql.com/doc/refman/5.7/en/type-conversion.html) is taking care of it.

File: modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc [hunk 2]
```diff
@@ -134,13 +134,22 @@ class NDB_Menu_Filter_Form_resolved_violations extends NDB_Menu_Filter_Form
...
         }
         else{
             $useProjects = true;
         }
+        // Get sites
+        $sitesCond = '';
+        if (!$user->hasPermission('access_all_profiles')
+            && !is_null($user->getData('CenterID'))
+        ) {
+            $sitesCond = " AND p.CenterID = '".$user->getData('CenterID')."' ";
```
Comment (xlecours) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_resolved_violations.class.inc]: Remark: p.CenterID is a numeric column and not text. The ' characters should be removed.
But MySQL is taking care of it.

**Comment by tedstrauss** [comment]
made changes suggested by @xlecours

File: modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc [hunk 3]
```diff
@@ -147,6 +147,13 @@ class NDB_Menu_Filter_Form_mri_violations extends NDB_Menu_Filter_Form
         else{
             $useProjects = true;
         }
+        // Get sites
+        $sitesCond = '';
+        if (!$user->hasPermission('access_all_profiles')
```
Comment (tedstrauss) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: choice of permission here may be up for debate.

Comment (driusan) [inline modules/mri_violations/php/NDB_Menu_Filter_Form_mri_violations.class.inc]: To clarify "choice of permission": There's also a permission named "violated_scans_view_allsites" in the schema.

**Comment by tedstrauss** [comment]
closing, will rebase for 17.0

**Pull request closed without merging**